cli/internal/cmd/apply.go

```diff
@@ -845,6 +806,42 @@ func (wl warnLogger) Warnf(fmtStr string, args ...any) {
 	wl.cmd.PrintErrf("Warning: %s\n", fmt.Sprintf(fmtStr, args...))
 }
 
+type warnLog interface {
+	Warnf(format string, args ...any)
+	Infof(format string, args ...any)
+}
+
+// applier is used to run the different phases of the apply command.
+type applier interface {
```

Comment:
I'm not a huge fan of having all interfaces at the end of this (very large) file. Personally, I'd rather have them below the struct / func they are used in